planner/core/planbuilder.go

```diff
@@ -1521,6 +1521,47 @@ func (b *PlanBuilder) buildDDL(node ast.DDLNode) (Plan, error) {
 			table:     v.Table.Name.L,
 			err:       nil,
 		})
+		for _, spec := range v.Specs {
+			if spec.Tp == ast.AlterTableRenameTable {
+				b.visitInfo = append(b.visitInfo, visitInfo{
+					privilege: mysql.DropPriv,
+					db:        v.Table.Schema.L,
+					table:     v.Table.Name.L,
+					err:       nil,
```

Comment:
`err` is used when privilege check failed. `"privilege check fail"` will be returned to the client if it is nil. How about setting the correct error code and message to make the error returned to the client also be compatible with MySQL?

Reply:
Not only here, if we wanna be consistent with mysql, need to fix all the places. Maybe we should create a new issue to fix them. But I think it is not the problem with this PR.

Reply:
https://github.com/pingcap/tidb/issues/9962

Reply:
Please use `appendVisitInfo()` to append a new visitinfo. see https://github.com/pingcap/tidb/issues/8946 for detail.

Reply:
@xiekeyi98 Please construct the correct error, below is an example:

```
var authErr error
	if b.ctx.GetSessionVars().User != nil {
		authErr = ErrTableaccessDenied.GenWithStackByArgs("SELECT", b.ctx.GetSessionVars().User.Hostname, b.ctx.GetSessionVars().User.Username, tableInfo.Name.L)
	}
	b.visitInfo = appendVisitInfo(b.visitInfo, mysql.SelectPriv, dbName.L, tableInfo.Name.L, "", authErr)
```

Reply:
Do I need to construct the error?
IMO, this PR is better to only fix the privilege check.
And we should better file another PR to fix the error compatibility(#9962).

Reply:
This PR is aimed to fix the incompatible issue, it's great that you have fixed the privilege check, but I think you can step further to fix the incompatible error message when privilege check failed in this PR.

As you can see in issue https://github.com/pingcap/tidb/issues/8946, we have fixed some of the incompatible error messages. Fixing the error message in this PR also saves the future work of others or yours.